core/modules/widgets/eventcatcher.js

```diff
@@ -77,6 +77,9 @@ EventWidget.prototype.render = function(parent,nextSibling) {
 			}
 			// Execute our actions with the variables
 			if(actions) {
+				if(variables === undefined) {
+					variables = $tw.utils.collectDOMVariables(selectedNode,self.domNode,event);
+				}
```

Comment:
Is this change related to the bug fix? We're already calling collectDOMVariables shortly before, which suggests that that call is in the wrong codepath.

Reply:
We're calling collectDOMVariables (which instantiates the "variables" variable) if there's a selector given.
If no selector is given we don't collect DOM variables and the "variables" variable is undefined at this point.
Now that I think about it better, it's probably the best to instantiate the `variables` variable at the top as an object and pass it to the `collectDOMVariables` method

Reply:
I've removed the second collection of the DOMVariables and fixed the instantiation of the variables variable